cpp/vcpkg.json

```diff
@@ -16,11 +16,16 @@
     },
     "benchmark",
     "boost-cmake",
+    "boost-beast",
     "boost-crc",
     "boost-filesystem",
+    "boost-locale",
     "boost-multiprecision",
+    "boost-optional",
     "boost-process",
     "boost-system",
+    "boost-variant",
+    "boost-xpressive",
```

Comment:
Note to myself: After this PR is merged, I will make comment on https://github.com/apache/arrow/pull/40939:

please review the `cpp/vcpkg.json` changes